src/integer_mod_q/mat_zq/from.rs

```diff
@@ -125,6 +125,41 @@ impl MatZq {
         }
         out
     }
+
+    /// Generate a `num_rows` times `num_columns` matrix with `1` on the
+    /// diagonal and `0` anywhere else.
```

Comment:
not true if modulus is `1`.

Please include that the matrix also has a modulus

Reply:
That case makes the function much more inefficient

Reply:
interesting (I did not expect the flint methods to not check that case)... I did not expect that behaviour to be honest. I did not expect it to be this way, but since this is the case, we probably have to life with it now :D